proposals/0079-allow_commission_decrease_at_any_time.md

```diff
@@ -0,0 +1,88 @@
+of next epoch, disallowing decreases during the second half of epoch is
+unnecessary.
+
+A feature gate must be added to support this SIMD as all validators' vote
+programs must treat commission change instructions the same or else consensus
+will diverge.
+
+## Motivation
+
+Some validator operators may need to decrease commission in order to satisfy
+their own operational criteria.
+
+As an example, a validaor operator may have a policy whereby any error that
+results in reduced stake account rewards for the epoch, will result in the
+operator choosing to reduce commission to 0% for that epoch to ensure that
+stake accounts are not disadvantaged by that error.  Not being allowed to do
+this in the second half of an epoch is a problem because it would prevent that
+commission change until the next epoch, which will not allow this policy to
+take effect for stake accounts which were de-activating during that epoch.
+
+Allowing commission changes during the second half of epochs also enables
+"reverse rug pulls" where a validator has commission set to 100% during the
+majority of an epoch but reduces it to 0% across the epoch boundary.  It's not
+clear why some validator operators choose to do this, but it does not seem
+harmful, and there does appear to be motivation for some validators to do this
+as it's been done numerous times in the past.
+
+## Alternatives Considered
+
+No alternatives were considered.
+
+## New Terminology
+
+Reverse Rug Pull: Setting commission to 100% for most of an epoch, and to 0%
+only across epoch boundaries.  This has the effect of discouraging third party
+stake to a validator while still charging 0% commission.  This is not
+considered to be a technique harmful to stakers.
+
+## Detailed Design
+
+A feature will be added which, when enabled, must cause all node
+implementations' vote program's set-commission instruction handling to first
+check whether the proposed commission change is a decrease or no change, and if
+so, do not invoke the "only allow commission change in first half of epoch"
+rule.
+
+## Impact
+
+Validators will now be able to decrease commission at any time in the epoch,
+but only increase commission in the first half of epochs (because of the
+commission_updates_only_allowed_in_first_half_of_epoch feature already
+implemented).
+
+## Security Considerations
+
+None
+
+## Drawbacks
+
+It may cause additional confusion to validators who might not understand why
+some types of commission changes succeed only in the first half of epochs while
+others succeed always.
+
+## Backwards Compatibility
+
+This feature requires a feature gate because software which includes the
+implementation will allow certain set-commission transactions to succeed where
+software without the implementation would fail those transactions.  Thus all
+validators must be updated to the new functionality at an epoch boundary so
+that all validators agree on the result of executing those transactions.
```

Comment:
OK

Reply:
So I'll include the note about the ledger, but I don't think I'll include the note about why it's fully backwards compatible for users of the stake program, since that isn't necessary in a section that is described (in the template) as " All incompatibilities and consequences should be listed."